packages/lodestar/src/sync/reqResp/reqResp.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @module sync
+ */
+
+import BN from "bn.js";
+import PeerInfo from "peer-info";
+import {
+  Epoch, Hash, Slot,
+  RequestBody, Hello, Goodbye,
+  BeaconBlocksRequest, BeaconBlocksResponse,
+  RecentBeaconBlocksRequest, RecentBeaconBlocksResponse,
+} from "@chainsafe/eth2.0-types";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+
+import {ZERO_HASH, Method, RequestId} from "../../constants";
+import {IBeaconDb} from "../../db";
+import {IBeaconChain} from "../../chain";
+import {INetwork} from "../../network";
+import {ReputationStore} from "../reputation";
+import {ILogger} from "../../logger";
+import {ISyncReqResp, ISyncOptions} from "./interface";
+
+export interface SyncReqRespModules {
+  config: IBeaconConfig;
+  db: IBeaconDb;
+  chain: IBeaconChain;
+  network: INetwork;
+  reps: ReputationStore;
+  logger: ILogger;
+}
+
+/**
+ * The SyncReqResp module handles app-level requests / responses from other peers,
+ * fetching state from the chain and database as needed.
+ */
+export class SyncReqResp implements ISyncReqResp {
+  private opts: ISyncOptions;
+  private config: IBeaconConfig;
+  private db: IBeaconDb;
+  private chain: IBeaconChain;
+  private network: INetwork;
+  private reps: ReputationStore;
+  private logger: ILogger;
+
+  public constructor(opts: ISyncOptions, {config, db, chain, network, reps, logger}: SyncReqRespModules) {
+    this.config = config;
+    this.opts = opts;
+    this.db = db;
+    this.chain = chain;
+    this.network = network;
+    this.reps = reps;
+    this.logger = logger;
+  }
+
+  private async createHello(): Promise<Hello> {
+    let headSlot: Slot,
+      headRoot: Hash,
+      finalizedEpoch: Epoch,
+      finalizedRoot: Hash;
+    if (!this.chain.isInitialized()) {
+      headSlot = 0;
+      headRoot = ZERO_HASH;
+      finalizedEpoch = 0;
+      finalizedRoot = ZERO_HASH;
+    } else {
+      headSlot = await this.db.chain.getChainHeadSlot();
+      const [bRoot, state] = await Promise.all([
+        this.db.chain.getBlockRoot(headSlot),
+        this.db.state.getLatest(),
+      ]);
+      headRoot = bRoot;
+      finalizedEpoch = state.finalizedCheckpoint.epoch;
+      finalizedRoot = state.finalizedCheckpoint.root;
+    }
+    return {
+      forkVersion: this.chain.latestState.fork.currentVersion,
+      finalizedRoot,
+      finalizedEpoch,
+      headRoot,
+      headSlot,
+    };
+  }
+
+  private handshake = async (peerInfo: PeerInfo): Promise<void> => {
+    const randomDelay = Math.floor(Math.random() * 5000);
+    await new Promise((resolve) => setTimeout(resolve, randomDelay));
+    if (
+      this.network.hasPeer(peerInfo) &&
+      !this.reps.get(peerInfo.id.toB58String()).latestHello
+    ) {
+      const request = await this.createHello();
+      await this.network.reqResp.hello(peerInfo, request).catch((e) => {});
```

Comment:
not handling the error here

Reply:
do we care if peer throws error on hello? I imagine it could happen quite often and not much you can do though

Reply:
We're suppressing the error here so the function doesn't throw.
This probably could be more robust, but this is the same implementation as before, just copy pastad from the old file.
Lets make an issue about this (also including handling incorrect forkVersion / disjoint finalizedCheckpoint in received hello messages) to make the handshake/hello process more robust and handle it in another PR. This way we can finish this big PR (and start testing it for interop) and go back to smaller bite-sized updates.